closure/templates/test/localized_test.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016 The Closure Rules Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+goog.require('goog.asserts');
+goog.require('goog.soy');
+goog.require('goog.testing.asserts');
+goog.require('goog.testing.jsunit');
+goog.require('io.bazel.rules.closure.soy.greeter');
+
+
+function testGreetLocalized() {
+  goog.soy.renderElement(goog.global.document.body,
+                         io.bazel.rules.closure.soy.greeter.greetLocalized,
+                         {name: 'Rob'});
+
+  var body = document.body;
+  goog.asserts.assert(body != null);
+  assertHTMLEquals('<p>Hello <b>Rob</b>!', body.innerHTML);
+}
```

Comment:
We ideally should test functionality of localization; ideally via replacing generated goog.getMsg calls somehow but this is an area of Closure that I'm not familiar at all so if you know how to do that it would be good.

Otherwise as an alternative, instead of having a closure test we can have a check that checks the output. Something like  following:

closure_js_template_library(
    name = "localized_soy",
    srcs = ["localized.soy"],
    should_generate_soy_msg_defs = True,
    bidi_global_dir = 1,
)

file_test(
    name = "localized_soy_generates_goog_get_msg",
    file = "localized.soy.js",
    regexp = "goog\.getMsg\(",
)

file_test(
    name = "non_localized_soy_not_generates_goog_get_msg",
    file = "greeter.soy.js",
    regexp = "goog\.getMsg\(",
    invert = True,
)

Reply:
I'm actually not familiar with that either. My particular usage replaces the goog.getMsg calls with a custom translation lookup. I just updated it with your latter suggestion.

Reply:
Do you really need to binary? Couldn't you just use the output of the closure_js_template_library target?

Reply:
Oh interesting, I didn't realize that closure/templates/test:localized.soy.js was a target I could build to get a standalone library file! Will do

Reply:
Updated